package/MDAnalysis/analysis/base.py

```diff
@@ -96,10 +92,14 @@ def run(self):
         self._prepare()
         for i, ts in enumerate(
                 self._trajectory[self.start:self.stop:self.step]):
-            self._ts = ts
```

Comment:
Is there a reason why this line has to go?  Could we call `_single_frame` with `ts` as the argument?
ie.
``` python
for i, ts in enumerate(self._traj[etc]):
    self._single_frame(ts)
```